pkg/filesystem/subvolume.go

```diff
@@ -52,6 +60,48 @@ func List(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace
 	listCephFSSubvolumes(ctx, clientsets, operatorNamespace, clusterNamespace, includeStaleOnly, subvolumeNames)
 }
 
+// checkForExternalStorage checks if the external mode is enabled.
+func checkForExternalStorage(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace string) bool {
+	enable := false
+	cephclusters, _ := clientsets.Rook.CephV1().CephClusters(clusterNamespace).List(ctx, v1.ListOptions{})
+	for _, cephcluster := range cephclusters.Items {
+		enable = cephcluster.Spec.External.Enable
```

Comment:
just return here of use break so that if we find an external cluster we don't need to go in the loop again.